src/app/static/src/app/store/baseline/mutations.ts

```diff
@@ -46,46 +50,48 @@ export const mutations: MutationTree<BaselineState> & BaselineMutations = {
         state.pjnzError = "";
     },
 
-    ShapeUpdated(state: BaselineState, action: PayloadWithType<ShapeResponse>) {
+    [BaselineMutation.ShapeUpdated](state: BaselineState, action: PayloadWithType<ShapeResponse>) {
         state.shape = Object.freeze(action.payload);
-        if (action.payload && action.payload.filters.regions){
+        if (action.payload && action.payload.filters.regions) {
             state.regionFilters = action.payload.filters.regions.children as NestedFilterOption[];
             state.flattenedRegionFilters = Object.freeze(flattenOptions(state.regionFilters));
         }
         state.shapeError = "";
     },
 
-    ShapeUploadError(state: BaselineState, action: PayloadWithType<string>) {
+    [BaselineMutation.ShapeUploadError](state: BaselineState, action: PayloadWithType<string>) {
         state.shapeError = action.payload;
     },
 
-    PopulationUpdated(state: BaselineState, action: PayloadWithType<PopulationResponse>) {
+    [BaselineMutation.PopulationUpdated](state: BaselineState, action: PayloadWithType<PopulationResponse>) {
         state.population = action.payload;
         state.populationError = "";
     },
 
-    PopulationUploadError(state: BaselineState, action: PayloadWithType<string>) {
+    [BaselineMutation.PopulationUploadError](state: BaselineState, action: PayloadWithType<string>) {
         state.populationError = action.payload;
     },
 
-    Validating(state: BaselineState){
+    [BaselineMutation.Validating](state: BaselineState) {
         state.validating = true;
         state.validatedConsistent = false;
         state.baselineError = "";
     },
 
-    Validated(state: BaselineState, action: PayloadWithType<ValidateBaselineResponse>) {
+    [BaselineMutation.Validated](state: BaselineState, action: PayloadWithType<ValidateBaselineResponse>) {
         state.validating = false;
 
         state.validatedConsistent = action.payload.consistent;
         state.baselineError = "";
     },
 
-    BaselineError(state: BaselineState, action: PayloadWithType<string>) {
+    [BaselineMutation.BaselineError](state: BaselineState, action: PayloadWithType<string>) {
         state.validating = false;
         state.validatedConsistent = false;
         state.baselineError = action.payload;
     },
 
-    ...readyStateMutations
+    [BaselineMutation.Ready](state: ReadyState) {
```

Comment:
It's a bit of a shame if this is going to have to be reimplemented in every module. Could we can a ReadyStateMutations enum and continue to pull in ...readyStateMutations here?

Reply:
I actually think this is better. It only gets repeated 3 times, it's only a couple of lines of code, and abstracting it into a shared class obscures what's happening a bit.